app/components/page-components/Chatbot/windows/ChatbotAlertsWindow.vue.ts

```diff
@@ -1,103 +1,108 @@
-      .keys(this.selectedTabData)
-      .find((key: string) => key.indexOf('messages') > -1);
-    const messages = this.selectedTabData[messageKey];
-    if (this.selectedTab === 'subscriptions') {
-      return messages;
-    }
-    if (this.selectedTab === 'followers') {
-      return {
-        all_alerts: messages.map((message: string) => ({message}))
+  get selectedTypeTableColumns() {
+    const message = this.selectedTypeMessages[0];
+    if (message) return Object.keys(message);
+
+    return [];
+  }
+
+  isEnabled(type: string) {
+    return this.alertTypes[type].enabled;
+  }
+
+  showNewChatAlertWindow() {
+    this.$modal.show('new-alert', {
+      onSubmit: (newAlert: any) => {
+        this.addNewAlert(this.selectedType, newAlert);
       }
-    }
-    return { all_alerts: messages };
+    });
   }
 
-  get selectedTabTableTitles() {
-    return Object.keys(this.selectedTabMessages);
+  onEdit(message: any, index: number) {
+    this.$modal.show('new-alert', {
+      editedAlert: message,
+      onSubmit: (updatedAlert: any) => {
+        this.spliceAlertMessages(this.selectedType, index, updatedAlert);
+      }
+    });
   }
 
-  get selectedTabTableColumns() {
-    const firstKey = this.selectedTabTableTitles[0];
-    const message = this.selectedTabMessages[firstKey][0];
-    if (message) return Object.keys(message);
+  onDelete(index: number) {
+    this.spliceAlertMessages(this.selectedType, index, null);
+  }
 
-    return [];
+  onDone() {
+    this.chatbotCommonService.closeChildWindow();
   }
 
+  // filters
+  formatNumber(value: any, dp?: number) {
+    if (isNaN(Number(value))) {
+      return value;
+    }
 
-  get tabs() {
-    const {
-      use_tip,
-      tip_messages
-    } = this.chatAlerts.settings.streamlabs;
-
-    const {
-      use_follow,
-      follow_messages,
-      use_host,
-      host_messages,
-      use_raid,
-      raid_messages,
-      use_sub,
-      subscriber_messages
-    } = this.chatAlerts.settings.twitch;
-
-    const tabs: AlertWindowData = {
-      followers: { use_follow, follow_messages },
-      subscriptions: { use_sub, subscriber_messages },
-      donations: { use_tip, tip_messages },
-      hosts: { use_host, host_messages } ,
-      raids: { use_raid, raid_messages },
+    if (dp === undefined) dp = 2;
+
+    return value.toLocaleString(undefined, {
+      maximumFractionDigits: dp,
+      minimumFractionDigits: dp
+    });
+  }
+
+  formatHeader(column: string) {
+    switch (column) {
+      case 'is_gifted':
+        return 'is gifted';
+      default:
+        return column;
     }
-    return tabs;
   }
 
-  onDone() {
-    this.chatbotCommonService.closeChildWindow();
+  formatValue(value: any, column: string) {
+    switch (column) {
+      case 'amount':
+        return this.formatNumber(value, 2);
+      case 'message':
+        return value;
+      case 'is_gifted':
+        return value === true ? 'Yes' : 'No';
+      default:
+        return value;
+    }
   }
 }
```

Comment:
It looks like this file in general makes heavy use of `any`.  It would be good to that cleaned up with proper interfaces/types.